utils/docker-ce/Config.in

```diff
@@ -0,0 +1,105 @@
+menu "Configuration"
+	depends on PACKAGE_docker-ce
+
+	config DOCKER_KERNEL_OPTIONS
+		bool "Enable Basic kernel support for Docker"
+		default y
+		select KERNEL_CGROUPS
+		select KERNEL_CGROUP_CPUACCT
+		select KERNEL_CGROUP_SCHED
+		select KERNEL_NAMESPACES
+		select KERNEL_CPUSETS
+		select KERNEL_MEMCG
+		select KERNEL_KEYS
+		select KERNEL_POSIX_MQUEUE
```

Comment:
I am afraid that this part requiring global kernel config change would be problematic.

When doing release builds
- If docker-ce was enabled on phase1, additional kernel features will be enabled by default for the whole build
- If docker-ce was only enabled later on phase2, the build could fail or it could succeed but fail to run for lacking kernel support.

Reply:
I'll see what I can do otherwise, but it looks like [LXC](https://github.com/openwrt/packages/blob/master/utils/lxc/Config.in#L4) just compiles without kernel support by default (and assumably fails at runtime), I've made changes if this would also suffice for docker-ce, for now.

Reply:
It seems that `KERNEL_CGROUPS` and `KERNEL_NAMESPACES` are unable to be loaded as a kernel module. Given that these are integral to containers/docker and nothing (at least from myself) can be done to resolve it, it will have to be compiled into the kernel. 

I assume that these features would just be bloat for anyone not wanting to run containers, so doubt they would be accepted in a PR in the main repository. Unless it were to be done like the LXC package, however, given these changes only affect the docker-ce package, this repository seems like a better fix.

Given that some if not all of these kernel configurations cannot be modularized, I see little benefit in looking into the other config options as they are only useful for the docker-ce package, at least currently.

If I am missing something in my Makefile to disable docker-ce for these phases you mentioned, or something so that the package is only explicitly enabled when desired, I am unaware and all ears.